client/client_test.go

```diff
@@ -2732,6 +2790,14 @@ func TestRotateKeyAfterPublishServerManagementChange(t *testing.T) {
 		data.CanonicalTargetsRole:  false,
 		data.CanonicalRootRole:     false,
 	})
+	// check that the snapshot remote rotation creates new keys
```

Comment:
the snapshot case is definitely tested in the previous case but I wanted to do an additional rotation from a pre-existing server-managed snapshot key.

I can add the check that those keys exist on disk to `testRotateKeySuccess`